### PR TITLE
OpenMC: add v0.13.1

### PR DESCRIPTION
**Pull request by paulromano**
This PR adds version 0.13.1 for packages openmc and py-openmc.

### CODE REVIEW

File: var/spack/repos/builtin/packages/openmc/package.py [hunk 1]
```diff
@@ -43,15 +45,21 @@ class Openmc(CMakePackage):
...
         options = ["-DCMAKE_INSTALL_LIBDIR=lib"]  # forcing bc sometimes goes to lib64
 
+        use_newer_options = self.spec.satisfies('@0.13.1:')
         if "+mpi" in self.spec:
             options += [
                 "-DCMAKE_C_COMPILER=%s" % self.spec["mpi"].mpicc,
                 "-DCMAKE_CXX_COMPILER=%s" % self.spec["mpi"].mpicxx,
             ]
+            if use_newer_options:
+                options += ["-DOPENMC_USE_MPI:BOOL=ON"]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openmc/package.py]:
```suggestion
                options += [self.define("OPENMC_USE_MPI", True)]
```

File: var/spack/repos/builtin/packages/openmc/package.py [hunk 2]
```diff
@@ -43,15 +45,21 @@ class Openmc(CMakePackage):
...
-        options += [self.define_from_variant("optimize")]
-        options += [self.define_from_variant("debug")]
+        if use_newer_options:
+            options += [self.define_from_variant("OPENMC_USE_OPENMP", "openmp")]
+        else:
+            options += [self.define_from_variant("openmp")]
+            options += [self.define_from_variant("optimize")]
+            options += [self.define_from_variant("debug")]
 
         if "+optimize" in self.spec:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openmc/package.py]: This looks really hacky and will lead to non-deterministic builds: `openmp build_type=RelWithDebInfo +optimize` and `openmp build_type=Debug +optimize` both install the same thing but have different hashes. I would remove the `+optimize` and `+debug` variants entirely and rely on the existing `build_type` variant found in all CMake packages.

File: var/spack/repos/builtin/packages/py-openmc/package.py [hunk 3]
```diff
@@ -31,7 +33,7 @@ class PyOpenmc(PythonPackage):
     variant("mpi", default=False, description="Enable MPI support")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-openmc/package.py:33]: I would remove this variant, it doesn't do anything in the package. If users want MPI support, they can do `spack install py-openmc ^openmc+mpi ^hdf5+mpi`.

**Comment by paulromano** [comment]
Thanks for the review @adamjstewart! I've just updated this branch based on your suggestions. Definitely simplifies things to remove the superfluous variants :+1:

File: var/spack/repos/builtin/packages/py-openmc/package.py (hunk 3 above)
Comment (paulromano) [reply var/spack/repos/builtin/packages/py-openmc/package.py:33]: One question regarding this -- how would I handle the dependency to `py-mpi4py`? Right now, when the `+mpi` variant is added, it picks up the `py-mpi4py` dependency. I'm not sure how to do that in the absence of the `mpi` variant within `py-openmc`.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-openmc/package.py:33]: Oh you're right, in that case I would keep the variant.

File: var/spack/repos/builtin/packages/openmc/package.py [hunk 4]
```diff
@@ -18,22 +18,21 @@ class Openmc(CMakePackage):
...
     version("master", branch="master", submodules=True)
+    version("0.13.1", commit="33bc948f4b855c037975f16d16091fe4ecd12de3", submodules=True)
     version("0.13.0", commit="cff247e35785e7236d67ccf64a3401f0fc50a469", submodules=True)
     version("0.12.2", commit="cbfcf908f8abdc1ef6603f67872dcf64c5c657b1", submodules=True)
     version("0.12.1", commit="36913589c4f43b7f843332181645241f0f10ae9e", submodules=True)
     version("0.12.0", commit="93d6165ecb455fc57242cd03a3f0805089c0e0b9", submodules=True)
     version("0.11.0", sha256="19a9d8e9c3b581e9060fbd96d30f1098312d217cb5c925eb6372a5786d9175af")
     version("0.10.0", sha256="47650cb45e2c326ae439208d6f137d75ad3e5c657055912d989592c6e216178f")
 
-    variant("mpi", default=False, description="Enable MPI support")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openmc/package.py:33]: This variant is used in many places, I wouldn't remove it

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**